common/src/main/scala/gu/msnotifications/Responses.scala

```diff
@@ -65,7 +66,11 @@ object RegistrationResponse {
   }
 }
 case class RegistrationResponse(registration: WNSRegistrationId, tags: List[String], channelUri: String, expirationTime: DateTime) {
-  def toRegistrarResponse : RegistrarResponse = ???
+  def toRegistrarResponse : RegistrarResponse = RegistrarResponse(
+    deviceId = channelUri,
+    WindowsMobile,
+    userId = Tags(tags.toSet).findUserId.get
```

Comment:
In that case, return Error \/ RegistrarResponse.  Encoding the possibility of error into the type forces the caller into handling the unexpected behaviour whereas a theoretical exception can be easily ignored